### PR TITLE
docs: Adding new documentation for library functions

### PR DESCRIPTION
**Pull request by ernestojeda**
Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/main/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Documentation content changes
- [ ] Other... Please describe:

## Sandbox Testing
Test Links :

### CODE REVIEW

File: vars/edgeXDocker.groovy [hunk 1]
```diff
@@ -14,6 +14,42 @@
...
+ - `edgeXDocker.generateDockerComposeForBuild`: Supporting function for `edgeXDocker.buildInParallel` that generates a docker compose file for a given list of docker images.
+ - `edgeXDocker.generateServiceYaml`: Supporting function for `edgeXDocker.buildInParallel` that generates service level yaml for a specific docker image.
+ - `edgeXDocker.push`: Push a specific docker image and optionally tag it with the `latest` tag. A nexus repository can also optionally be specified as well as specific tags.
+ - `edgeXDocker.pushAll`: Push all docker images specified in the `dockerImages` list. To be used in conjunction with the same input format used by `edgeXDocker.buildInParallel` to push all images.
+ - `edgeXDocker.getDockerTags`: Generates the default set of tags used when pushing all edgex docker images with the `edgeXDocker.push` function.
+ - `edgeXDocker.finalImageName`: Never used due to issues with embedded version labels in images.
+ - `edgeXDocker.relabel`: Never used to the complexity of relabeling images.
+ - `edgeXDocker.finalImageName`: Prepends a docker image with `env.DOCKER_REGISTRY_NAMESPACE` if defined.
```
Comment (soda480) [inline vars/edgeXDocker.groovy:35]: `edgexDocker.finalImageName` is noted twice.

File: vars/edgeXDocker.groovy [hunk 2]
```diff
@@ -14,6 +14,42 @@
...
+ - `edgeXDocker.buildInParallel`: Build multiple docker images in parallel. This technique utilizes docker-compose to build multiple images using the parallel flag.
+ - `edgeXDocker.generateDockerComposeForBuild`: Supporting function for `edgeXDocker.buildInParallel` that generates a docker compose file for a given list of docker images.
+ - `edgeXDocker.generateServiceYaml`: Supporting function for `edgeXDocker.buildInParallel` that generates service level yaml for a specific docker image.
+ - `edgeXDocker.push`: Push a specific docker image and optionally tag it with the `latest` tag. A nexus repository can also optionally be specified as well as specific tags.
+ - `edgeXDocker.pushAll`: Push all docker images specified in the `dockerImages` list. To be used in conjunction with the same input format used by `edgeXDocker.buildInParallel` to push all images.
+ - `edgeXDocker.getDockerTags`: Generates the default set of tags used when pushing all edgex docker images with the `edgeXDocker.push` function.
+ - `edgeXDocker.finalImageName`: Never used due to issues with embedded version labels in images.
+ - `edgeXDocker.relabel`: Never used to the complexity of relabeling images.
```
Comment (soda480) [inline vars/edgeXDocker.groovy:34]: should `edgexDocker.relabel` be removed since the function is removed?

Comment (ernestojeda) [reply vars/edgeXDocker.groovy:34]: Yes. Will remove

File: vars/edgeXDocker.groovy (hunk 1 above)
Comment (ernestojeda) [reply vars/edgeXDocker.groovy:35]: Thanks. I will fix this.

File: vars/edgeXDocker.groovy [hunk 3]
```diff
@@ -14,6 +14,42 @@
...
+ - `edgeXDocker.parse`: Reads a docker image url and returns the parsed [image object](#parsed-image-object) components
+ - `edgeXDocker.toImageStr`: Returns docker image string from an [image object](#parsed-image-object)
+
+ ## Parsed image object
+
+ ```groovy
+ [
+    host: hostname if any associated with the image,
+    fullImage: full image name with tag,
+    namespace: namespace of the image is any,
```
Comment (bill-mahoney) [inline vars/edgeXDocker.groovy:46]: "**if** any"

**Review by bill-mahoney — DISMISSED** [review]
Minor typo, but looks good to me.

**Review by soda480 — APPROVED** [review]
LGTM

**Pull request merged**